### PR TITLE
Added the contributing guide

### PR DESCRIPTION
**Pull request by hoichi**
Here's something to start, at the long last.

A couple of questions:

* What should multi-word hashtags look like? Hyphens? Low dashes? Nodelimitersatall?
* Do we accept multi-type content? For instance, an online version of Eloquent JavaScript is both a #book and #exercises. Although I don't know if there's a lot of other examples like this.

### CODE REVIEW

**Comment by ericelliott** [comment]
> What should multi-word hashtags look like? Hyphens? Low dashes? Nodelimitersatall?

Underscore

> Do we accept multi-type content? For instance, an online version of Eloquent JavaScript is both a #book and #exercises. Although I don't know if there's a lot of other examples like this.

Yes. One link can have several hashtags.

File: doc/Contributing Guide.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
+# Contributing Guide
+
+These resources will be used by JavaScript developers of all levels to learn the language itself, the best programming practices and the essential libraries, frameworks and tools that are part of the modern JS scene.
+
+We're collecting free training resources that can either be used as self-teaching materials or as a reference for quality materials on certain topics. Noone can possibly learn all the tools and libraries in existence — not all at once, at the very least, but can certainly need a place to look for a library or tool that solves a particular problem.  
+
+One of the reason for this program is to make these materials available for use by housing-first job training programs for the homeless, but there's no reason not to use them by anyone interested.
```
Comment (ericelliott) [inline doc/Contributing Guide.md]: "One of the reason" should be "One of the reasons".

File: doc/Contributing Guide.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
+# Contributing Guide
+
+These resources will be used by JavaScript developers of all levels to learn the language itself, the best programming practices and the essential libraries, frameworks and tools that are part of the modern JS scene.
+
+We're collecting free training resources that can either be used as self-teaching materials or as a reference for quality materials on certain topics. Noone can possibly learn all the tools and libraries in existence — not all at once, at the very least, but can certainly need a place to look for a library or tool that solves a particular problem.
```
Comment (ericelliott) [inline doc/Contributing Guide.md]: > Noone can possibly learn all the tools and libraries in existence — not all at once, at the very least, but can certainly need a place to look for a library or tool that solves a particular problem.

This bit is self evident. Delete.

Comment (ericelliott) [inline doc/Contributing Guide.md]: On second thought, delete this entire paragraph.

File: doc/Contributing Guide.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+* Exercises. Just like tutuorials, but exercises tell the student when something is wrong and let them try to fix it. e.g. [learnrx](http://jhusain.github.io/learnrx/) **#exercises**
+* Project. A resource that describes a project for a student and lets the student complete it. **#project**
+* Game. A game designed to teach the student how to do something. **#game**
+* Infographic. A graphical visualization of information, somtimes connecting several data sets to reveal a larger picture. **#infographic**
+* Community. An online place for students to meet other people interested in the topic so they can learn from each other. **#community**
+* News. A resource that lets the student track current developments about the topic. **#news**
+* Collection. An expert-curated collection of resources for a particular topic (like this one!). **#collection**
+* Course. A full course curriculum, typically put together for an educational organization such as a university, or online school. **#course**
+* Tool. A code editor, a builder, a linter, a debug suite and so on. **#tool**
+
```
Comment (ericelliott) [inline doc/Contributing Guide.md]: Add Library as `#library`

File: doc/Contributing Guide.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+* [Why JavaScript?](https://vimeo.com/101269995) by [Eric Elliott](https://twitter.com/_ericelliott) #video
+* [JavaScript for Cats](https://medium.com/javascript-scene/learn-javascript-b631a4af11f2) by [Max Ogden](http://jsforcats.com/) #tutorial #vanilla #beginner
+* [Eloquent JavaScript](http://eloquentjavascript.net/) by [Marijn Haverbeke](marijnjh) #book
+
+## Example Reference Collection
+
+### Transpilers
+
+* [Babel](https://github.com/babel/babel) Transpile ES6, ES7 to ES5. #tool #ES6 #ES2015
+* [CoffeeScript](http://coffeescript.org/) The very concise language that compiles to JS. #tool #languages
```
Comment (ericelliott) [inline doc/Contributing Guide.md]: Use the singular, #language

File: doc/Contributing Guide.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+* [Emscripten](http://kripken.github.io/emscripten-site/) Frequently used for C/C++ JavaScript ports. #tool #languages #C
```
Comment (ericelliott) [inline doc/Contributing Guide.md]: ditto

**Comment by hoichi** [comment]
Updated the branch.

**Comment by ericelliott** [comment]
If we move this to the root and call it `contributing.md` GitHub will automatically detect it and suggest that people read it when they open a new issue. =)

**Pull request merged**